internal/usecase/devices/alarms.go

```diff
@@ -83,16 +88,64 @@ func (uc *UseCase) addAlarmOutputEntityToDTO(d *amtAlarmClock.AddAlarmOutput) *d
 	return d1
 }
 
-func (uc *UseCase) alarmOccurenceEntityToDTO(d *alarmclock.AlarmClockOccurrence) *dto.AlarmClockOccurrence {
-	startTime, _ := time.Parse(time.RFC3339, d.StartTime)
-	interval, _ := strconv.Atoi(d.Interval)
+func (uc *UseCase) alarmOccurrenceEntityToDTO(d *alarmclock.AlarmClockOccurrence) *dto.AlarmClockOccurrence {
+	intervalInMinutes, _ := ParseInterval(d.Interval.Interval)
+	interval, _ := strconv.Atoi(d.Interval.Interval)
 	d1 := &dto.AlarmClockOccurrence{
 		ElementName:        d.ElementName,
 		InstanceID:         d.InstanceID,
-		StartTime:          startTime,
+		StartTime:          d.StartTime.Datetime,
 		Interval:           interval,
+		IntervalInMinutes:  intervalInMinutes,
 		DeleteOnCompletion: d.DeleteOnCompletion,
 	}
 
 	return d1
 }
+
+func ParseDuration(dur *string, unit byte) (int, error) {
+	if idx := strings.IndexByte(*dur, unit); idx != -1 {
+		part, err := strconv.Atoi((*dur)[:idx])
+		if err != nil {
+			return 0, err
+		}
+
+		*dur = (*dur)[idx+1:]
+
+		return part, nil
+	}
+
+	return 0, nil
+}
+
+func ParseInterval(duration string) (int, error) {
```

Comment:
did you look at the time package provided from go? you should be able to use time.ParseDuration to simplify all this a bit.

Reply:
done